### PR TITLE
Deprecate sk_types.h and move pscontext class to a separate file pscontext.h

### PR DESCRIPTION
**Pull request by jeffli-xilinx**
#### Problem solved by the commit
Move pscontext class out of sk_types.h and make sk_types.h to be deprecated

### CODE REVIEW

**Review by stsoe — APPROVED** [review]

File: src/runtime_src/core/edge/include/sk_types.h [hunk 1]
```diff
@@ -30,7 +31,8 @@
  * freeBO    : free BO handle.
  * logMsg    : send log messages to XRT driver for saving as per ini settings
  */
-struct sk_operations {
+struct XRT_DEPRECATED
```
Comment (sonals) [inline src/runtime_src/core/edge/include/sk_types.h:34]: Should the `XRT_DEPRECATED` come before `struct`?

File: src/runtime_src/core/edge/include/sk_types.h [hunk 2]
```diff
@@ -45,29 +47,11 @@ struct sk_operations {
...
-
-struct pscontext::impl {
-private:
-  bool aie_profile_en;
-};
+#pragma message ("sk_types.h is deprecated and will be removed from the distribution directory in a future release.  Please use pscontext.h instead.")
 
-typedef pscontext* (* kernel_init_t)(xclDeviceHandle device, const uuid_t &uuid);
-typedef int (* kernel_fini_t)(pscontext *xrtHandles);
+#include "pscontext.h"
```
Comment (sonals) [inline src/runtime_src/core/edge/include/sk_types.h:55]: Can add a comment here saying this include is for backward compatibility?

**Review by sonals — CHANGES_REQUESTED** [review]
Minor changes

File: src/runtime_src/core/edge/include/sk_types.h (hunk 1 above)
Comment (jeffli-xilinx) [reply src/runtime_src/core/edge/include/sk_types.h:34]: I had it before struct, but got warning when compiling saying attribute ignored, and it need to follow struct keyword

File: src/runtime_src/core/edge/include/sk_types.h (hunk 2 above)
Comment (jeffli-xilinx) [reply src/runtime_src/core/edge/include/sk_types.h:59]: Comment added

File: src/runtime_src/core/edge/include/sk_types.h (hunk 1 above)
Comment (sonals) [reply src/runtime_src/core/edge/include/sk_types.h:34]: Thanks for detail.

**Review by sonals — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]

**Pull request merged**